### PR TITLE
add com.github.shonumi.gbe-plus

### PR DESCRIPTION
**Pull request by hadess**
TODO:

- [X] Finish installation and startup scripts
- [X] Create .desktop file
- [X] Create .appdata.xml file
- [X] Test

### CODE REVIEW

File: com.github.shonumi.gbe-plus.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    "finish-args" : [
+        /* X11 + XShm access */
+        "--share=ipc", "--socket=x11",
+        /* GPU acceleration */
+        "--device=dri",
+        /* Play sounds */
+        "--socket=pulseaudio",
+        /* Get access to the files */
+        "--filesystem=host:ro",
+        "--filesystem=xdg-run/gvfs:ro"
```
Comment (TingPing) [inline com.github.shonumi.gbe-plus.json]: Is GVFS used by Qt?

File: io.qt.qt5-base.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
+{
```
Comment (TingPing) [inline io.qt.qt5-base.json]: This was mentioned to you already but you should just use the KDE runtime. Their build of Qt has a dozen Flatpak specific patches and is actually well maintained by Qt contributors. This is not something we want packages to reproduce.

File: com.github.shonumi.gbe-plus.json (hunk 1 above)
Comment (hadess) [reply com.github.shonumi.gbe-plus.json]: No, I didn't update the finish-args in that version.

File: io.qt.qt5-base.json (hunk 2 above)
Comment (hadess) [reply io.qt.qt5-base.json]: It's using the KDE runtime now, as I really don't want to maintain Qt on my own indeed.

**Comment by hadess** [comment]
Only problem seems to be that ROMs don't open when using File/Open, which is a bit weird. For example:
```
The specified file: '/run/user/1000/doc/5bac5a86/Wario Land - Super Mario Land 3 (World).gb' could not be loaded
```
The file lives in ~/Documents/Games on my system.

File: local-build.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+rm -f com.github.shonumi.gbe-plus.flatpak
```
Comment (bilelmoussaoui) [inline local-build.sh]: No need to have this file here.

Comment (hadess) [reply local-build.sh]: I like it, and have something similar in all my Flathub projects.

**Comment by nedrichards** [comment]
bot, build com.github.shonumi.gbe-plus

**Comment by hadess** [comment]
bot, build com.github.shonumi.gbe-plus

**Comment by grulja** [comment]
So I checked this, it looks that QFile::exists() doesn't correctly report whether the file exists. It actually exists and is opened just fine if I ignore this check. I don't know how the flatpak document portal exposes the files for opening, whether it's a symlink or something like that, but you can use QFileInfo::exists(QString::fromStdString(config::rom_file)) in that condition instead, this seems to work as it should. Another thing which will be causing problems for you will be saving probably, it automatically tries to save the game progress into "config::rom_file.sav", which won't be writtable, the save file needs to be picked through the portal as well.

**Comment by hadess** [comment]
> So I checked this, it looks that QFile::exists() doesn't correctly report whether the file exists.

Any chance you could file a bug against Qt about that? Sounds like a problem we might encounter in the future as well.

> Another thing which will be causing problems for you will be saving probably, it automatically tries to save the game progress into "config::rom_file.sav", which won't be writtable, the save file needs to be picked through the portal as well.

That's a known problem for a number of applications, we'll need a directory portal of some kind, or a modification on the document portal to solve that.

**Comment by grulja** [comment]
> > So I checked this, it looks that QFile::exists() doesn't correctly report whether the file exists.
> 
> Any chance you could file a bug against Qt about that? Sounds like a problem we might encounter in the future as well.
> 
I'll try to investigate why it fails and open a bug in case it's their fault.

**Comment by nedrichards** [comment]
@hadess do you want to hold this until there's an xdg-desktop-portal release with that commit? (I know there's a couple of other TODOs, but just checking)

**Comment by grulja** [comment]
I think it's better to go with the previous workaround, using QFileInfo::exists() for now, because it doesn't mean that when xdg-desktop-portal gets released, that it will be available everywhere.

**Comment by hadess** [comment]
> @hadess do you want to hold this until there's an xdg-desktop-portal release with that commit? (I know there's a couple of other TODOs, but just checking)

No, I switched around to sharing the whole home, as we need access to the directories to save progress anyway. I think I'm just missing the appdata file now.

**Comment by hadess** [comment]
Ready to go!
Do we have a better solution for the screenshots being missing during review, but are going to be in the repo once it's made?

**Comment by nedrichards** [comment]
bot, build com.github.shonumi.gbe-plus

**Comment by nedrichards** [comment]
@hadess ARM failure looks like it could be a GL vs GLES issue? Could just disable on there if it's too much bother.

**Comment by nedrichards** [comment]
bot, build com.github.shonumi.gbe-plus

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.github.shonumi.gbe-plus

**Comment by nedrichards** [comment]
@hadess anyone else to add?

**Comment by hadess** [comment]
Maybe @shonumi as the upstream?

**Pull request closed without merging**